### ISSUE TITLE
Possible false positive with UP006 and UP007

### ISSUE THREAD

**Issue by sbdchd** [charliermarsh/ruff#1617]
I'm not sure if this is a false positive or if I shouldn't enable these rules

I like using the newest syntax in `.pyi` files (enforced by UP006, UP007) but I have a project running on 3.7 so it's mostly `List[str]` instead of `list[str]` style things

But, `from __future__ import annotations` should allow for newer annotations, in practice this project uses pydantic which uses the python builtins to parse the annotations at runtime, so they have to be runtime compatible otherwise python explodes:

```
______________________________________________________________ ERROR collecting kodiak/tests/evaluation/test_merge_message.py ______________________________________________________________
kodiak/tests/evaluation/test_merge_message.py:7: in <module>
    from kodiak.config import (
kodiak/config.py:57: in <module>
    class AutomergeDependencies(BaseModel):
.venv/lib/python3.7/site-packages/pydantic/main.py:188: in __new__
    annotations = resolve_annotations(namespace.get('__annotations__', {}), namespace.get('__module__', None))
.venv/lib/python3.7/site-packages/pydantic/typing.py:419: in resolve_annotations
    value = _eval_type(value, base_globals, None)
../../../.pyenv/versions/3.7.13/lib/python3.7/typing.py:263: in _eval_type
    return t._evaluate(globalns, localns)
../../../.pyenv/versions/3.7.13/lib/python3.7/typing.py:467: in _evaluate
    eval(self.__forward_code__, globalns, localns),
<string>:1: in <module>
    ???
E   TypeError: 'type' object is not subscriptable
```

This might be more of a "don't use those rules unless you're on newer versions of python"

**Comment by charliermarsh** [comment]
Does this do what you're looking for?

```toml
[tool.ruff.pyupgrade]
keep-runtime-typing = true
```

**Comment by sbdchd** [comment]
Oh yeah that works!

I need to RTFM

Thank you!

**Comment by charliermarsh** [comment]
Not at all, the docs are a little unwieldy right now. They have a lot of good info, but it's just not that discoverable. It's on my mind, working on it...

If you're coming from existing tools, I do try to pull in the most useful settings verbatim (for example, `pyupgrade` has this same setting, under this same name). Maybe helpful in the future. Although if you've never used `pyupgrade`, then this advice is totally useless :)
